bam_plcmd.c

```diff
@@ -554,12 +556,16 @@ static int mpileup(mplp_conf_t *conf, int n, char **fn)
     // init pileup
     iter = bam_mplp_init(n, mplp_func, (void**)data);
     if ( conf->flag & MPLP_SMART_OVERLAPS ) bam_mplp_init_overlaps(iter);
-    max_depth = conf->max_depth;
-    if (max_depth * sm->n > 1<<20)
-        fprintf(stderr, "(%s) Max depth is above 1M. Potential memory hog!\n", __func__);
-    if (max_depth * sm->n < 8000) {
-        max_depth = 8000 / sm->n;
-        fprintf(stderr, "<%s> Set max per-file depth to %d\n", __func__, max_depth);
+    if ( !conf->max_depth ) {
+        max_depth = INT_MAX / sm->n;
+    } else {
+        max_depth = conf->max_depth;
+        if ( max_depth * sm->n > 1<<20 )
+            fprintf(stderr, "(%s) Max depth is above 1M. Potential memory hog!\n", __func__);
+        if ( max_depth < MPLP_MIN_DEPTH ) { 
```

Comment:
This isn't quite the same as before.

If we have 1000 samples and asked for max depth of 10 then previously that's what we'd have got as 10 * 1000 >= 8000. If we'd have set it to 2 then it'd have bumped it to 8 for us.  This is a bit strange as setting it to 8000 say (the default) actually means 8000 * 1000 total aggregate depth.  So sometimes the code is per sample and sometimes it's total.  I think this is likely a mistake.

Now we're doing the more logical thing which is a per sample limit that applies equally regardless of whether we have 1 or 1000 samples; it does what the user expects.  However if they have 10,000 samples they may well wish to set max_depth (per sample) to just 50 say.  I think what we have is correct, the new minimum value is perhaps too high.

I can't really judge what is appropriate though.  @pd3 care to comment?

Reply:
In bcftools the original code was changed to this https://github.com/samtools/bcftools/blob/develop/mpileup.c#L607-L610, so I am assuming this was inspired by it.

Note that in bcftools this is only a warning. The value 250 is the number of reads that contribute to the calculation of genotype likelihoods, but the code there does not enforce the limit in case the user wants to override it.